src/audio/module_adapter/module/volume/volume_generic_with_peakvol.c

```diff
@@ -144,6 +149,7 @@ static void vol_s32_to_s32(struct processing_module *mod, struct input_stream_bu
 							   Q_SHIFT_BITS_64(31, VOL_QXY_Y, 31));
 				tmp = MAX(abs(x0[i]), tmp);
 			}
+			tmp = tmp << attenuation;
```

Comment:
I missed this review. But why isn't this max value scaling done at this function calling level?